command/base_flags.go

```diff
@@ -30,6 +30,86 @@ type FlagBool interface {
 	IsBoolFlag() bool
 }
 
+// BoolPtr is a bool which is aware if it has been set.
+type BoolPtr struct {
+	v *bool
+}
+
+func (b *BoolPtr) Set(v string) error {
+	if b.v == nil {
+		b.v = new(bool)
+	}
+	var err error
+	*(b.v), err = strconv.ParseBool(v)
```

Comment:
When there's a possible error It's usually preferable to use a temporary value rather than assigning directly to the target.  If ParseBool returns an error we'll be overwriting b.v, possibly setting a default-true BoolPtr to be default-false.

Reply:
Yep. Fixed.